bagit.xml

```diff
@@ -825,654 +806,432 @@ myfirstbag/
-
-<t>
-Some cautions regarding bag interchange arise in regard to the
-commonly available checksum tools distributed with the GNU Coreutils
-package (md5sum, sha1sum, etc.), collectively referred to here as
-"md5sum".  First, md5sum can be run in binary or text
-mode; text mode sometimes normalizes line-endings.  While these
-modes appear to produce the same checksums under Unix-like systems, they
-can produce different checksums under Windows.  When using md5sum, it 
-may be safest to run it in binary mode, with one caveat:  a side-effect
-of binary mode is that md5sum requires a space and an asterisk ('*'),
-compared to two spaces in text mode, between the CHECKSUM and FILENAME in
-its manifest format.
-</t>
-
-<t>
-Due to the widespread use of md5sum (and its relatives), it is not
-unexpected for bag receivers to see manifests in which CHECKSUM and
-FILENAME are separated by a space followed by an asterisk.  Implementors
-creating or processing bags with md5sum should be aware of these subtle
-differences, and ensure compliance with the manifest specification in this
-document.  Implementors creating and processing bags with other tools may wish
-to be tolerant of asterisks found in the manifests.
-</t>
-
-<t>A final note about md5sum-generated manifests is that for a
-FILENAME containing a backslash ('\'), the manifest line will have a
-backslash inserted in front of the CHECKSUM and, under Windows, the
-backslashes inside FILENAME may be doubled.
-</t>
-
-</section>
-
-<section title="Windows and Unix file naming">
+        </figure>
+      </section>
+    </section>
+    <!-- /Examples -->
+    <section title="Security Considerations" anchor="sec-security">
+      <section title="Special directory characters">
+        <t>
+          The paths specified in the payload manifests, tag manifests, and
+          fetch files do not prohibit special directory characters which have
+          special meaning on some operating systems. Implementers &must; ensure
+          that files outside the bag directory structure are not accessed when
+          reading or writing files based on paths specified in a bag.
+        </t>
+        <t>
+          All implementations &should; have a test suite to guard against
+          special directory characters.
+        </t>
+        <t>
+          For example, a maliciously crafted "tagmanifest-md5.txt" file might
+          contain entries which begin with a path character such as "/", "..",
+          or a "~username" home directory reference in an attempt to cause a
+          naive implementation to leak or overwrite targeted files on a POSIX
+          operating system.
+        </t>
+        <t>
+          Windows implementations &should; test their implementations to ensure
+          that safety-checks prevent use of drive letters and the less commonly used
+          namespace sequences (e.g. "\\?\C:\&#x2026;") described in <xref target="MSFNAM"/>.
```

Comment:
As well as [Alternative Data Streams](https://blogs.technet.microsoft.com/askcore/2013/03/24/alternate-data-streams-in-ntfs/) which could be used to hide files (including executables) inside another file entry.

Reply:
That raises some other questions — that's in the Windows section but most other operating systems have similar concepts (extended attributes or resource forks) and there are non-trivial portability concerns trying to use them. 

Technically this could currently be considered a client implementation issue – e.g. a Windows bag creation tool could have manifests listing `foo.txt`, `foo.txt:$DATA`, etc. but that'd definitely increase the odds of someone experiencing compatibility issues moving that to other systems.

I'm inclined to say that BagIt should discourage all of this in favor of portability and have a suggestion that if you need to preserve non-portable filesystem metadata you create an archive using the platform's preferred format and bag that.

Reply:
The Alternative Data Streams are commonly used for attacks such as hiding root kits. Several [CVEs](https://cwe.mitre.org/data/definitions/69.html) have been raised on this.

So if someone maliciously mints a bag with such "files" in it, when extracted they would disappear from regular listing in the filesystem (but can still be executed). 

Perhaps this is more of a problem for `fetch.txt` than the manifest - of course if the bag is in an archive this is a problem with a naive unarchiving process, the malicious user would not really need to list it in the manifest (unless the implementation is stringently checking against its listing during extraction) as most likely the "complete" check of the archive would not check for ADS-es.

Reply:
I think this is an implementation issue — and e.g. many places require virus scanning — but it might be useful to put a note about that along with the other points in the implementers guide since we already have other notes about security and filesystem quirks. Checking for alternate data streams seems like a good area to encourage awareness around since they're not uncommon.